### PR TITLE
Cleanup and refactoring

### PR DESCRIPTION
**Pull request by ruflin**
I was going through the code base to get more familiar with it. Along the way I made some small changes / cleanups that I stumbled over. There are also parts where I added small tests which are more for me to understand the code on what it does. Here is the list:

* Remove savedObjectKey function as it is not needed anymore
 * add config options for monitoring to the default configuration for documentation purpose
* move the creation of the nowString to where it is actually used. Seems it is used only once.
* add test for convertActions to better understand the format on my end
* Add note to missing parts
* update t.Fatal should not be used in test go routine
* add basic token test
* add more entries to the config file about available config options
* remove unused env package
* remove not used `missing.txt` file.
* remove dsl readme and filed #136 instead

### CODE REVIEW

File: cmd/fleet/missing.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# TODO: Can we convert this to Github issues, still needed?
```
Comment (ruflin) [inline cmd/fleet/missing.txt:1]: @ph Seems you created this file. I think we should convert these to gibhub issues. We have stats, what about the others?

Comment (ph) [reply cmd/fleet/missing.txt:1]: I think this come from the original directory when I've closed @scunningham's repository. 
I am not sure what "run as admin; doesn't drop creds" mean? is that dropping privileges?

Comment (scunningham) [reply cmd/fleet/missing.txt:1]: A bit cryptic on my part.  That was a note to myself that there is no reason for the fleet server to run with Elastic admin privileges.  We are addressing that with service accounts.

**Review by scunningham — APPROVED** [review]
Can remove the internal env package as well. No longer used.

**Comment by ruflin** [comment]
Unused env package removed.

File: cmd/fleet/handleEnroll.go [hunk 2]
```diff
@@ -347,6 +346,7 @@ func decodeEnrollRequest(data io.Reader) (*EnrollRequest, error) {
 
 	// Validate
 	switch req.Type {
+	// TODO: Should these be converted to constant? Need to be kept in sync with Kibana?
```
Comment (ruflin) [inline cmd/fleet/handleEnroll.go:349]: @nchaulet @blakerouse Thoughts?

File: cmd/fleet/missing.txt (hunk 1 above)
Comment (ruflin) [reply cmd/fleet/missing.txt]: I removed it.

**Comment by ruflin** [comment]
Marked as ready for review. No real changes in this PR. More similar PR's might follow when I dig further into the code.

File: cmd/fleet/handleEnroll.go (hunk 2 above)
Comment (blakerouse) [reply cmd/fleet/handleEnroll.go:349]: @nchaulet Is this even used?

**Review by blakerouse — APPROVED** [review]
Clean ups look good.

File: cmd/fleet/handleEnroll.go (hunk 2 above)
Comment (nchaulet) [reply cmd/fleet/handleEnroll.go:349]: No it's not probably no reason to keep it :)

Comment (ruflin) [reply cmd/fleet/handleEnroll.go:349]: Not sure what the paths forward here is. Can the validation part be removed?  I assume the decode we still need.

Comment (nchaulet) [reply cmd/fleet/handleEnroll.go:349]: I think we could make the `type` field optional in Kibana (we already do not use it), remove it from the agent, and remove it from fleet server.

Comment (ruflin) [reply cmd/fleet/handleEnroll.go:349]: I did not touch it for now and merged the PR. It sounds removing it might break things in Agent or Fleet.

**Pull request merged**